test/linter-test.js

```diff
@@ -445,4 +445,14 @@ describe('linter', () => {
     assert.calledOnceWithMatch(callback, 'formatters/unknown');
   });
 
+  it('sets exit code of caught error to 2', async () => {
```

Comment:
I think this test belongs to the `withinDirectory` tests, so that all versions are tested.

Reply:
Got it. I moved the test into the `withinDirectory` block with a slight modification depending on the ESLint version, since I couldn't get the same error to throw with all versions.